purchase_order_supplierinfo_update/models/purchase_order.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def write(self, vals):
+        no_updated = self.filtered(lambda r: r.state not in ["purchase", "done"])
+        res = super().write(vals)
+        if vals.get("state", "") in ["purchase", "done"]:
+            no_updated.mapped("order_line").update_supplierinfo_price()
+        return res
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res.update_supplierinfo_price()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("price_unit"):
+            self.update_supplierinfo_price()
```

Comment:
In a confirmed purchase order, if you change the prices, it will update the supplier_info. It is also possible to evaluate the discount.

```suggestion
        if vals.get("price_unit") or "discount" in vals:
            self.update_supplierinfo_price()
```